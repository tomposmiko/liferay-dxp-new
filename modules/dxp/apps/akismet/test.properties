##
## Test Batch
##

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]=\
        (testray.main.component.name ~ "Akismet")

##
## Testray
##

    testray.main.component.name=Akismet