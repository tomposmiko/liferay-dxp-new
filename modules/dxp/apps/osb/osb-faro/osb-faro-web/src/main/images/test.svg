<svg fill="none" height="20" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg">
	<rect fill="#6B6C7E" height="20" rx="5" width="20" />
	<path d="M14.7653 13.6478C13.308 10.8283 11.8837 9.2493 11.8762 7.75V5.5C11.8762 5.5 12.1619 5.5007 12.2512 5.5C13.2184 5.4918 13.2937 4.0382 12.2512 4H7.75116C6.72389 4.05789 6.77545 5.46133 7.75116 5.5H8.12616V7.75C8.13975 9.25164 6.56006 11.0352 5.23702 13.6478C4.92225 14.2696 5.09452 15.8924 6.99202 16H13.0103C14.6774 15.9386 15.1429 14.3784 14.7653 13.6478ZM12.9285 14.5H7.05577C6.84577 14.5 6.57342 14.2649 6.74077 13.9075L7.37499 12.6252C9.27108 12.6252 9.97022 11.1231 11.5103 11.1241C11.5103 11.1241 13.1483 13.784 13.2058 13.8625C13.4507 14.1979 13.2137 14.5 12.9285 14.5Z" fill="white" />
</svg>