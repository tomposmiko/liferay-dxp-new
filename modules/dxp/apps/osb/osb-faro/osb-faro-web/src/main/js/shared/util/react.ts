import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
