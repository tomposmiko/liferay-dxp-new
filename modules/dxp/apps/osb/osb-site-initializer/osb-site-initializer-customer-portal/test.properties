##
## Test Batch
##

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer Customer Portal") OR \
            (testray.main.component.name ~ "OSB Site Initializer Customer Portal")\
        )