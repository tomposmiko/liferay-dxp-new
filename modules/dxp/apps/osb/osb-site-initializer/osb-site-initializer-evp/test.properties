##
## Test Batch
##

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer EVP") OR \
            (testray.main.component.name ~ "OSB Site Initializer EVP")\
        )

##
## Testray
##

    testray.main.component.name=OSB Site Initializer EVP