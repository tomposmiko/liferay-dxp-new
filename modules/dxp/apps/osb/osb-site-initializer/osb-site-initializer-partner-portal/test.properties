##
## Salesforce
##

    #site.initializer.partner.portal.login.url=
    #site.initializer.partner.portal.consumer.key=
    #site.initializer.partner.portal.consumer.secret=
    #site.initializer.partner.portal.username=
    #site.initializer.partner.portal.password=

##
## Test Batch
##

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer Partner Portal") OR \
            (testray.main.component.name ~ "OSB Site Initializer Partner Portal")\
        )

##
## Testray
##

    testray.main.component.name=OSB Site Initializer Partner Portal