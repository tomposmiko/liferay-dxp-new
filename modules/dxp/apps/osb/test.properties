##
## Testray
##

    testray.main.component.name=Customer Portal