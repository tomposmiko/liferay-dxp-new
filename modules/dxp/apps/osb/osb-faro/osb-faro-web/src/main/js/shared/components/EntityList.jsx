import autobind from 'autobind-decorator';
import Avatar from './Avatar';
import getCN from 'classnames';
import ListGroup from './list-group';
import ListView from './ListView';
import omitDefinedProps from 'shared/util/omitDefinedProps';
import React from 'react';
import Spinner from './Spinner';
import Sticker from './Sticker';
import TextTruncate from './TextTruncate';
import {EntityTypes, SegmentTypes} from '../util/constants';
import {getDataSourceLangKey} from 'shared/util/lang';
import {getRouteName, Routes, toRoute} from 'shared/util/router';
import {LIFERAY_SITE_TYPE} from 'shared/util/data-sources';
import {Link} from 'react-router-dom';
import {PropTypes} from 'prop-types';
import {Set} from 'immutable';
import {sub} from '../util/lang';

class EntityListItem extends React.Component {
	static propTypes = {
		item: PropTypes.object.isRequired
	};

	state = {
		checked: false
	};

	getEntityUrl() {
		const {
			channelId,
			groupId,
			item: {id, type, url}
		} = this.props;

		switch (type) {
			case LIFERAY_SITE_TYPE:
				return url ? url : '#';
			case EntityTypes.DataSource:
				return toRoute(Routes.SETTINGS_DATA_SOURCE, {
					groupId,
					id
				});
			default:
				return toRoute(Routes.CONTACTS_ENTITY, {
					channelId,
					groupId,
					id,
					type: getRouteName(type)
				});
		}
	}

	getMessage() {
		const {friendlyURL, properties, providerType, type} = this.props.item;

		switch (type) {
			case EntityTypes.Individual:
				return sub(Liferay.Language.get('x-at-x'), [
					properties.jobTitle ||
						Liferay.Language.get('not-available'),
					properties.worksFor || Liferay.Language.get('not-available')
				]);
			case EntityTypes.DataSource:
				return getDataSourceLangKey(providerType);
			case LIFERAY_SITE_TYPE:
				return friendlyURL;
			default:
				return;
		}
	}

	handleLinkClick(event) {
		event.stopPropagation();
	}

	render() {
		const {item} = this.props;

		const {name, segmentType, type} = item;

		return (
			<>
				<ListGroup.ItemField>
					{type === EntityTypes.IndividualsSegment ? (
						<Sticker
							display='light'
							symbol={
								segmentType === SegmentTypes.Static
									? 'individual-static-segment'
									: 'individual-dynamic-segment'
							}
						/>
					) : (
						<Avatar
							circle={type === EntityTypes.Individual}
							entity={item}
						/>
					)}
				</ListGroup.ItemField>

				<ListGroup.ItemField className='justify-content-center' expand>
					<TextTruncate title={name}>
						<strong>
							<Link
								className='list-group-link'
								onClick={this.handleLinkClick}
								to={this.getEntityUrl()}
							>
								{name ? name : Liferay.Language.get('unknown')}
							</Link>
						</strong>
					</TextTruncate>

					<div className='secondary-info'>{this.getMessage()}</div>
				</ListGroup.ItemField>
			</>
		);
	}
}

class EntityList extends React.Component {
	static defaultProps = {
		disabledItemsISet: new Set(),
		loading: false,
		selectedItemsISet: new Set(),
		selectMultiple: true,
		showBorder: true,
		showHeader: true
	};

	static propTypes = {
		accentColor: PropTypes.string,
		channelId: PropTypes.string,
		disabledItemsISet: PropTypes.instanceOf(Set),
		entityType: PropTypes.number,
		groupId: PropTypes.string.isRequired,
		header: PropTypes.string,
		items: PropTypes.array,
		loading: PropTypes.bool,
		noItemsContent: PropTypes.any,
		noItemsHeader: PropTypes.string,
		onSelectItemsChange: PropTypes.func,
		selectedItemsISet: PropTypes.instanceOf(Set),
		selectMultiple: PropTypes.bool,
		showBorder: PropTypes.bool,
		showHeader: PropTypes.bool,
		total: PropTypes.number
	};

	@autobind
	getItemRenderer() {
		return ({item}) => {
			const {channelId, groupId} = this.props;

			return (
				<EntityListItem
					channelId={channelId}
					groupId={groupId}
					item={item}
				/>
			);
		};
	}

	render() {
		const {
			className,
			disabledItemsISet,
			items,
			loading,
			noItemsContent,
			onSelectItemsChange,
			selectedItemsISet,
			selectMultiple,
			showBorder,
			total,
			...otherProps
		} = this.props;

		const noItems = total === 0;

		const classes = getCN('entity-list-root', className, {
			'no-items': noItems
		});

		return (
			<div className={classes}>
				<ListView
					{...omitDefinedProps(otherProps, EntityList.propTypes)}
					disabledItemsISet={disabledItemsISet}
					itemRenderer={this.getItemRenderer()}
					items={items}
					noBorder={!showBorder}
					onSelectItemsChange={onSelectItemsChange}
					selectedItemsISet={selectedItemsISet}
					selectMultiple={selectMultiple}
				/>

				{loading && <Spinner overlay />}

				{!loading && noItems && (
					<div className='status-overlay'>{noItemsContent}</div>
				)}
			</div>
		);
	}
}

export default EntityList;
