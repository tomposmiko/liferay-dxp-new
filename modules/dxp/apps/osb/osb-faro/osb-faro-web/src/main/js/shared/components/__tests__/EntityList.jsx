import EntityList from '../EntityList';
import React from 'react';
import {cleanup, render} from '@testing-library/react';
import {mockIndividual, mockSegment} from 'test/data';
import {withStaticRouter} from 'test/mock-router';

jest.unmock('react-dom');

const items = [
	mockIndividual(0, {total: 123}),
	mockIndividual(1, {total: 123}),
	mockIndividual(2, {total: 123})
];

const segments = [
	mockSegment(0, {segmentType: 'STATIC'}),
	mockSegment(1, {segmentType: 'DYNAMIC'}),
	mockSegment(2, {segmentType: 'STATIC'})
];

const WrappedComponent = withStaticRouter(EntityList);

describe('EntityList', () => {
	afterEach(cleanup);

	it('should render', () => {
		const {container} = render(
			<WrappedComponent groupId='23' items={items} />
		);

		expect(container).toMatchSnapshot();
	});

	it('should render noItemsContent', () => {
		const noItemsContent = <h1>{'No Items!'}</h1>;
		const {queryByText} = render(
			<WrappedComponent
				groupId='23'
				noItemsContent={noItemsContent}
				total={0}
			/>
		);

		expect(queryByText('No Items!')).toBeTruthy();
	});

	it('should render a list of segments with segment icons', () => {
		const {container} = render(
			<WrappedComponent groupId='23' header='foo bar' items={segments} />
		);

		expect(
			container.querySelectorAll(
				'.lexicon-icon-individual-static-segment'
			).length
		).toBe(2);

		expect(
			container.querySelectorAll(
				'.lexicon-icon-individual-dynamic-segment'
			).length
		).toBe(1);
	});
});
