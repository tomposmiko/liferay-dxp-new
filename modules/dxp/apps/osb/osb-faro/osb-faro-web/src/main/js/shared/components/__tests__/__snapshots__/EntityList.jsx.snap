// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityList should render 1`] = `
<div>
  <div
    class="entity-list-root"
  >
    <ul
      class="list-group list-group-root show-quick-actions-on-hover"
    >
      <li
        class="list-group-item list-group-item-flex"
      >
        <div
          class="autofit-col"
        >
          <div
            class="sticker sticker-root sticker-static avatar-root sticker-circle sticker-chartRed"
          >
            FB
          </div>
        </div>
        <div
          class="autofit-col justify-content-center autofit-col-expand"
        >
          <span
            class="text-truncate"
            data-tooltip="false"
            title=""
          >
            <strong>
              <a
                class="list-group-link"
                href="/workspace/23/contacts/individuals/0"
              >
                Foo Bar
              </a>
            </strong>
          </span>
          <div
            class="secondary-info"
          >
            Developer at N/A
          </div>
        </div>
      </li>
      <li
        class="list-group-item list-group-item-flex"
      >
        <div
          class="autofit-col"
        >
          <div
            class="sticker sticker-root sticker-static avatar-root sticker-circle sticker-chartSeaGreen"
          >
            FB
          </div>
        </div>
        <div
          class="autofit-col justify-content-center autofit-col-expand"
        >
          <span
            class="text-truncate"
            data-tooltip="false"
            title=""
          >
            <strong>
              <a
                class="list-group-link"
                href="/workspace/23/contacts/individuals/1"
              >
                Foo Bar
              </a>
            </strong>
          </span>
          <div
            class="secondary-info"
          >
            Developer at N/A
          </div>
        </div>
      </li>
      <li
        class="list-group-item list-group-item-flex"
      >
        <div
          class="autofit-col"
        >
          <div
            class="sticker sticker-root sticker-static avatar-root sticker-circle sticker-chartBlue"
          >
            FB
          </div>
        </div>
        <div
          class="autofit-col justify-content-center autofit-col-expand"
        >
          <span
            class="text-truncate"
            data-tooltip="false"
            title=""
          >
            <strong>
              <a
                class="list-group-link"
                href="/workspace/23/contacts/individuals/2"
              >
                Foo Bar
              </a>
            </strong>
          </span>
          <div
            class="secondary-info"
          >
            Developer at N/A
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
